src/config.rs

```diff
@@ -198,7 +208,42 @@ fn get_config_paths(filename: &str) -> Vec<PathBuf> {
             p
         })
     })
-    .collect()
+    .collect::<Vec<_>>();
+
+    paths.extend(dirs);
+    paths
```

Comment:
It could be also better here to extend `dirs` with `path`, instead of the other way around. I'd like for `RUSTIC_HOME` to be the top most.

Reply:
Yes, I think is best to create a vec of possible paths which are already ordered and then `filter_map` and `collect` as already done.

About the order, I agree: first the `RUSTIC_HOME` dir - and the additional windows dirs I actually would but last...

Reply:
Fixed (I think?)

Reply:
I think you should `extend` before `filter_map`...

Reply:
It's different types, win compatibility is already `Vec<PathBuf>` whereas the other paths are `Vec<Option<PathBuf>>`, hence the filter_map.

Reply:
```rust
#[cfg(target_os = "windows")]
fn get_windows_portability_config_directories() -> Option<Vec<PathBuf>> {
    std::env::var_os("USERPROFILE").map(|path| {
        vec![
            PathBuf::from(path.clone()).join(r".config\rustic"),
            PathBuf::from(path).join(".rustic"),
        ]
    })
}
```

Reply:
The cruical part is about the `map` which adds the profile filename to the path.
Maybe just change the function such that it returns `Vec<Option<PathBuf>>`?